sunkit_image/reproject.py

```diff
@@ -0,0 +1,106 @@
+"""
+Functionality for reprojecting solar images between different coordinate
+systems.
+"""
+import numpy as np
+from reproject import reproject_adaptive, reproject_exact, reproject_interp
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+from astropy.wcs import WCS
+from sunpy.map import Map, make_fitswcs_header
+from sunpy.sun import constants
+
+__all__ = ["reproject_carrington"]
+
+_reproject_funcs = {"interp": reproject_interp, "exact": reproject_exact, "adaptive": reproject_adaptive}
+
+
+def reproject_carrington(smap, shape_out, latitude_projection="CAR", *, reproject_method="interp", **reproject_kwargs):
+    """
+    Reproject *smap* into a Carrington frame of reference.
+
+    Parameters
+    ----------
+    smap : sunpy.map.GenericMap
+        Input map.
+    shape_out : [int, int]
+        Number of output pixels in (latitude, longitude).
+    latitude_projection : {"CAR", "CEA"}
+        Projection in the latitude direction, either plate-Carée or cylindrical-equal-area.
+    reproject_method : {"interp", "exact", "adaptive"}
+        Method to use for reprojection.
+    **reproject_kwargs
+        Additional keyword arguments to pass to the reproject function.
+        See the reproject docs for more info.
+
+    Returns
+    -------
+    carrington_map : sunpy.map.Genericmap
+        Reprojected map.
+    """
+    # The Carrington coordinate frame in sunpy assumes the standard photospheric
+    # radius for the surface of the sun, so need to temporarily set this in the
+    # map header
```

Comment:
`HeliographicCarrington` has supported the `rsun` frame attribute for a while now, and `make_fitswcs_header()` should populate `RSUN_REF` using that frame attribute.  This juggling of `RSUN_REF` should be unnecessary.  You should simply pass the `RSUN_REF` into your `carrington_header()` function to use when creating the reference coordinate.